File: app/src/main/java/co/za/takeabyte/gradalot2020/promotions/viewmodel/ViewModelPromotions.java
package co.za.takeabyte.gradalot2020.promotions.viewmodel;

import android.os.CountDownTimer;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;

import co.za.takeabyte.gradalot2020.promotions.uimodel.UIModelPromotionItem;
import co.za.takeabyte.gradalot2020.repository.IRepositoryPromotions;
import co.za.takeabyte.gradalot2020.repository.impl.RepositoryPromotions;

public class ViewModelPromotions extends ViewModel {

    private IRepositoryPromotions repository = new RepositoryPromotions();
    private MutableLiveData<List<UIModelPromotionItem>> promotions = new MutableLiveData<>();

    public LiveData<List<UIModelPromotionItem>> getPromotions() {
        return promotions;
    }

    public void fetchPromotions() {
        new CountDownTimer(5000, 1000) {

            @Override
            public void onTick(long millisUntilFinished) {
            }

            @Override
            public void onFinish() {
                promotions.postValue(repository.getPromotions());
            }
        }.start();
    }
}

File: app/src/main/java/co/za/takeabyte/gradalot2020/promotions/viewmodel/ViewModelPromotions.kt
package co.za.takeabyte.gradalot2020.promotions.viewmodel

import android.os.CountDownTimer
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import co.za.takeabyte.gradalot2020.promotions.uimodel.UIModelPromotionItem

class ViewModelPromotions : ViewModel() {
    val promotions: MutableLiveData<List<UIModelPromotionItem>> = MutableLiveData()

    fun getPromotions() {
        object : CountDownTimer(5000, 1000) {
            override fun onFinish() {
                promotions.postValue(generateDummyData())
            }

            override fun onTick(millisUntilFinished: Long) {}
        }.start()
    }

    private fun generateDummyData() = ArrayList<UIModelPromotionItem>().apply {
        add(
            UIModelPromotionItem(
                title = "GoldAir - 40cm Oscillating Fan - Black",
                price = 8999.00
            )
        )
        add(
            UIModelPromotionItem(
                title = "GoldAir - 40cm Oscillating Fan - Black",
                price = 8999.00
            )
        )
        add(
            UIModelPromotionItem(
                title = "GoldAir - 40cm Oscillating Fan - Black",
                price = 8999.00
            )
        )
        add(
            UIModelPromotionItem(
                title = "GoldAir - 40cm Oscillating Fan - Black",
                price = 8999.00
            )
        )
        add(
            UIModelPromotionItem(
                title = "GoldAir - 40cm Oscillating Fan - Black",
                price = 8999.00
            )
        )
        add(
            UIModelPromotionItem(
                title = "GoldAir - 40cm Oscillating Fan - Black",
                price = 8999.00
            )
        )
        add(
            UIModelPromotionItem(
                title = "GoldAir - 40cm Oscillating Fan - Black",
                price = 8999.00
            )
        )
        add(
            UIModelPromotionItem(
                title = "GoldAir - 40cm Oscillating Fan - Black",
                price = 8999.00
            )
        )
    }
}